### PR TITLE
refactor: refactor repeated `delete` calls 

### PR DESCRIPTION
**Pull request by yuktea**
### What
Refactored some repeated` delete `calls in:
`Display.pm`
`Ingredients.pm`
`Food.pm`

There are more of these but a lot of them are already in loops, so a nested loop really wouldn't make anything better.

### CODE REVIEW

**Review by yuktea — COMMENTED** [review]
I believe the jobs are failing because of no double quotes

**Comment by yuktea** [comment]
> I believe the jobs are failing because of no double quotes

The checks pass now!

File: lib/ProductOpener/Display.pm [hunk 1]
```diff
@@ -10329,9 +10329,10 @@ sub display_structured_response($)
 		# remove the languages field which has keys like "en:english"
 		# keys with the : character break the XML export
 
-		delete $request_ref->{structured_response}{product}{languages};
-		delete $request_ref->{structured_response}{product}{category_properties};
-		delete $request_ref->{structured_response}{product}{categories_properties};
+		my @foo = ("languages", "category_properties", "categories_properties");
```
Comment (stephanegigandet) [inline lib/ProductOpener/Display.pm:10332]: Instead create a function that we can call like this:

delete_fields($product_ref, ["languages", "category_properties", "categories_properties"]);

Comment (yuktea) [reply lib/ProductOpener/Display.pm]: Done! Looks much righter now.

Comment (yuktea) [reply lib/ProductOpener/Display.pm]: Thanks for the feedback! just made the changes

**Pull request merged**